### PR TITLE
Driver independence

### PR DESCRIPTION
**Pull request by wingo**
This PR is a reworking of #1199 which tries to first adapt network functions to be agnostic as far as whether they are using the old intel10g driver or the new intel_mp driver.  It does this by adding `rx` and `tx` fields to the PCI driver info structure, letting network functions know what link names to use.  It renames the `ndescriptors` property of `intel_mp` to `ring_buffer_size`, and adds a corresponding initarg to the old intel10g driver.  Finally it enables VMDq whereever network functions were setting a MAC address; see #749.

I have a few more unpushed commits to make corresponding changes to the lwAFTR and the changes that @teknico and @takikawa made to the packetblaster / firehose, and the commit that actually changes over the `pci.device_info()` function to prefer `intel_mp`, but every patch in this series should work and cause no regressions, so it should be a safe merge on all downstreams of `wingo-next` (including `lwaftr`).

### CODE REVIEW

File: src/apps/intel_mp/intel_mp.lua [hunk 1]
```diff
@@ -569,12 +570,14 @@ function Intel:push ()
    if not self.txq then return end
    local li = self.input["input"]
    if li == nil then return end
---   assert(li, "intel_mp:push: no input link")
 
    while not empty(li) and self:ringnext(self.tdt) ~= self.tdh do
       local p = receive(li)
-      -- NB: see comment in intel10g for why this is commented out,
-      --     the rest of the loop body goes in an else branch
+      -- NB: the comment below is taken from intel_mp.lua, which disables
```
Comment (teknico) [inline src/apps/intel_mp/intel_mp.lua]: Should this say `intel10g` instead of `intel_mp.lua`?

**Comment by wingo** [comment]
I am having trouble with the upstream bots for some reason; will PR this against `Igalia/snabb` to see what those bots think.

File: src/apps/intel_mp/intel_mp.lua (hunk 1 above)
Comment (wingo) [reply src/apps/intel_mp/intel_mp.lua]: Yes you are right.  Will fix in a followup.  Thanks for the review!

**Comment by wingo** [comment]
There is a test failure due to an `intel_mp` change that's already upstream, for intel1g cards only.  Merging this code as it's unrelated and we'll fix the intel1g problem later.

**Pull request merged**